modules/tensor_mechanics/src/materials/ComputeInterfaceStress.C

```diff
@@ -18,51 +18,79 @@ validParams<ComputeInterfaceStress>()
-  params.addRequiredParam<Real>("stress", "Planar stress");
-  params.addRangeCheckedParam<Real>("op_range",
-                                    1.0,
-                                    "op_range > 0.0",
-                                    "Range over which order parameters change across an "
-                                    "interface. By default order parameters are assumed to "
-                                    "vary from 0 to 1");
-  params.addParam<MaterialPropertyName>(
-      "planar_stress_name", "extra_stress", "Material property name for the planar stress");
+  params.addRequiredParam<std::vector<Real>>("stress",
+                                             "Interfacial planar stress magnitude (one "
+                                             "value to apply to all order parameters or one value "
+                                             "per order parameter listed in 'v')");
+  params.addRangeCheckedParam<std::vector<Real>>(
+      "op_range",
+      {1.0},
+      "op_range > 0.0",
+      "Range over which order parameters change across an "
+      "interface. By default order parameters are assumed to "
+      "vary from 0 to 1");
+  params.addParam<MaterialPropertyName>("planar_stress_name",
+                                        "extra_stress",
+                                        "Material property name for the interfacial planar stress");
   return params;
 }
 
 ComputeInterfaceStress::ComputeInterfaceStress(const InputParameters & parameters)
   : Material(parameters),
-    _grad_v(coupledGradient("v")),
-    _stress(getParam<Real>("stress") / getParam<Real>("op_range")),
+    _nvar(coupledComponents("v")),
+    _grad_v(_nvar),
+    _op_range(getParam<std::vector<Real>>("op_range")),
+    _stress(getParam<std::vector<Real>>("stress")),
     _planar_stress(
         declareProperty<RankTwoTensor>(getParam<MaterialPropertyName>("planar_stress_name")))
 {
+  if (_stress.size() == 1)
+    _stress.assign(_nvar, _stress[0]);
+  if (_stress.size() != _nvar)
+    mooseError("Supply either one single stress or one per order parameter");
+
+  if (_op_range.size() == 1)
+    _op_range.assign(_nvar, _op_range[0]);
+  if (_op_range.size() != _nvar)
+    mooseError("Supply either one single op_range or one per order parameter");
+
+  for (auto i = beginIndex(_grad_v); i < _nvar; ++i)
+  {
+    _grad_v[i] = &coupledGradient("v", i);
+    _stress[i] /= _op_range[i];
+  }
 }
 
 void
 ComputeInterfaceStress::computeQpProperties()
 {
   auto & S = _planar_stress[_qp];
+  S.zero();
 
-  // no interface, return zero stress
-  const Real grad_norm_sq = _grad_v[_qp].norm_sq();
-  if (grad_norm_sq < libMesh::TOLERANCE)
+  // loop over interface variables
+  for (auto i = beginIndex(_grad_v); i < _nvar; ++i)
   {
-    S.zero();
-    return;
-  }
+    // no interface, return zero stress
+    const Real grad_norm_sq = (*_grad_v[i])[_qp].norm_sq();
+    if (grad_norm_sq < libMesh::TOLERANCE)
+      continue;
```

Comment:
What's the purpose of this if statement and continue statement?

Reply:
"no interface, return zero stress" :-)

It means if the norm of the order parameter gradient is zero (within numerical tolerance) there is no interface to contribute to an interfacial planar stress

Reply:
I get that, but the lines themselves are extraneous, nothing is being done...

Reply:
That is incorrect

Reply:
@amjokisaari If nothing was being done and you just removed those lines, then look what would happen on line 83:
```
const Real s = _stress[i] / std::sqrt(grad_norm_sq);
```

But, I would agree that the word "return" in the comment is misleading...

Reply:
Ok, so I had to look up continue statements and I stand corrected.  Still don't like the construction of hiding the condition in an if and then breaking out of the loop, but I'm not going to bother considering if there's an alternative way to construct this.

Reply:
I hope the updated comment clarifies this

Reply:
okay.  The comment is better, I see what's going on now.